luajit.krun

```diff
@@ -0,0 +1,96 @@
+import os
+from krun.vm_defs import (LuaVMDef)
+from krun import EntryPoint
+from krun.util import fatal
+
+# Who to mail
+MAIL_TO = []
+
+# Maximum number of error emails to send per-run
+#MAX_MAILS = 2
+
+DIR = os.getcwd()
+
+HEAP_LIMIT = 2097152  # KiB
+STACK_LIMIT = 8192  # KiB
+
+# Variant name -> EntryPoint
+VARIANTS = {
+    "default-lua": EntryPoint("bench.lua"),
+}
+
+ITERATIONS_ALL_VMS = 5  # Small number for testing.
+
+VMS = {
+    'Normal': {
+        'vm_def': LuaVMDef("builds/luajit"),
+        'variants': ['default-lua'],
+        'n_iterations': ITERATIONS_ALL_VMS,
+    },
+    'GC64': {
+        'vm_def': LuaVMDef("builds/luajit_gc64"),
+        'variants': ['default-lua'],
+        'n_iterations': ITERATIONS_ALL_VMS,
+    },
+    'NoJIT': {
+        'vm_def': LuaVMDef("builds/luajit_nojit"),
+        'variants': ['default-lua'],
+        'n_iterations': ITERATIONS_ALL_VMS,
+    },
+    'DualNum': {
+        'vm_def': LuaVMDef("builds/luajit_dualnum"),
+        'variants': ['default-lua'],
+        'n_iterations': ITERATIONS_ALL_VMS,
+    },
+}
+
+BENCHMARKS = {
+    "binarytrees" : 1,
+    "nbody" : 1,
+    "fasta": 1,
+    "richards": 1,
+    "spectralnorm": 1,
+    "fannkuch_redux": 1,
+    "md5": 1,
+    "series": 1,
+    "luacheck_parser": 1,
+    "luacheck": 1,
+    "capnproto_encode": 1,
+    "capnproto_decode": 1,
+    "jsonlua_encode": 1,
+    "jsonlua_decode": 1,
+}
+
+# list of "bench:vm:variant"
+SKIP=[
+    #"*:C:*",
+    #"*:PyPy:*",
+]
+
+N_EXECUTIONS = 1  # Number of fresh processes.
+
+# No. of seconds to wait before taking the initial temperature reading.
+# You should set this high enough for the system to cool down a bit.
+# The default (if omitted) is 60 seconds.
+TEMP_READ_PAUSE = 2
+
+# Commands to run before and after each process execution
+#
+# Environment available for these commands:
+#   KRUN_RESULTS_FILE: path to results file.
+#   KRUN_LOG_FILE: path to log file.
+#   KRUN_ETA_DATUM: time the ETA was computed
+#   KRUN_ETA_VALUE: estimated time of completion
+#PRE_EXECUTION_CMDS = ["sudo service cron stop"]
+#POST_EXECUTION_CMDS = ["sudo service cron start"]
+
+# CPU pinning (off by default)
```

Comment:
Are you sure you want pinning? Is LuaJIT multi-threaded?

Reply:
I thought it could switch around cores from context switches but maybe I misunderstanding what it does

Reply:
You are right, that's what it's supposed to prevent, but you'd have to make sure luajit spins up an appropriate number of threads.

If you pin on a n-core machine, you will get a n-1 core "cset shield" to run on. You'd need to check that luajit doesn't spin up 4 cores worth of threads, as that will slow you down and distort your results (the jvm gets this wrong for example, so we gave up on pinning for the warmup experiment).

Reply:
If you are going to use regular lua, you'd need to do a similar check for fairness.

Reply:
LuaJIT is not multi-threaded in anyway except its sampled based profiler.

Reply:
Let's not use pinning please. It opens up a new world of pain.

Reply:
If LuaJIT and the reference implementation are truly single threaded, then pinning would be OK. Does that change your opinion Laurie?

Reply:
Nope :) I don't think we understand the effects of pinning well enough to make it worth the bother. One day, maybe we will know enough to be confident that turning it on is a good idea.

Reply:
OK